modules/base/dashboard/dashboarditemtimevaryingtext.cpp

```diff
@@ -0,0 +1,161 @@
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/dashboard/dashboarditemtimevaryingtext.h>
+
+#include <fstream>
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <openspace/engine/globals.h>
+#include <openspace/util/timemanager.h>
+#include <openspace/json.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/logging/logmanager.h>
+
+
+
+namespace {
+    constexpr std::string_view _loggerCat = "DashboardItemTimeVaryingText";
+
+    constexpr openspace::properties::Property::PropertyInfo TextBeforeInfo = {
+        "TextBefore",
+        "Text Before",
+        "Optional text before the time varying text to be displayed.",
+        openspace::properties::Property::Visibility::User
+    };
+    constexpr openspace::properties::Property::PropertyInfo DataFileInfo = {
+        "DataFile",
+        "Data File Path",
+        "The file path to the JSON data.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    struct [[codegen::Dictionary(DashboardItemTimeVaryingText)]] Parameters {
+        // [[codegen::verbatim(TextBeforeInfo.description)]]
+        std::optional<std::string> textBefore;
+        // [[codegen::verbatim(DataFileInfo.description)]]
+        std::string dataFile;
+    };
+#include "dashboarditemtimevaryingtext_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation DashboardItemTimeVaryingText::Documentation() {
+    return codegen::doc<Parameters>(
+        "base_dashboarditem_timevaryingtextBefore",
+        DashboardTextItem::Documentation()
+    );
+}
+
+DashboardItemTimeVaryingText::DashboardItemTimeVaryingText(
+                                                      const ghoul::Dictionary& dictionary)
+    : DashboardTextItem(dictionary)
+    , _textBefore(TextBeforeInfo, "")
+    , _dataFile(DataFileInfo, "")
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+    _textBefore = p.textBefore.value_or(_textBefore);
+    addProperty(_textBefore);
+    _dataFile = absPath(p.dataFile).string();
+    addProperty(_dataFile);
+    loadDataFromJson(_dataFile);
```

Comment:
This loading should happen in the `initialize` method so that it is multithreaded